oldschnorr.go

```diff
@@ -0,0 +1,149 @@
+package secp256k1
+
+// #include "./depend/secp256k1/include/secp256k1_oldschnorr.h"
+import "C"
+import "github.com/pkg/errors"
+
+// SchnorrPublicKey is a PublicKey type used to sign and verify Schnorr signatures.
+// The struct itself is an opaque data type that should only be created via the supplied methods.
+type SchnorrPublicKey struct {
+	pubkey C.secp256k1_pubkey
+}
+
+// SchnorrSignature is a type representing a Schnorr Signature.
+// The struct itself is an opaque data type that should only be created via the supplied methods.
+type SchnorrSignature struct {
+	signature [64]byte
+}
+
+// Serialize returns a 64 byte serialized signature
+func (signature *SchnorrSignature) Serialize() [64]byte {
+	return signature.signature
+}
+
+// DeserializeSchnorrSignature deserializes a 64 byte serialized schnorr signature into a SchnorrSignature type.
+func DeserializeSchnorrSignature(serializedSignature [64]byte) SchnorrSignature {
+	signature := SchnorrSignature{}
+	signature.signature = serializedSignature
```

Comment:
You can write this whole function in one line `return SchnorrSignature{signature:serializedSignature}`